packages/graphql/src/schema-model/Operation.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Neo4jGraphQLSchemaValidationError } from "../classes";
+
+import type { Annotation, Annotations } from "./annotation/Annotation";
+import { annotationToKey } from "./annotation/Annotation";
+import type { Field } from "./attribute/Field";
+
+export class Operation {
+    public readonly name: string;
+    // custom cypher fields
+    public readonly fields: Map<string, Field> = new Map();
```

Comment:
If this needs a comment should the naming not just reflect that it's custom cypher fields?

```suggestion
    public readonly customCypherFields: Map<string, Field> = new Map();
```

Reply:
For now it is custom Cypher fields only, but this could also include fields resolved by custom resolvers. I think sticking with GraphQL nomenclature here is correct - operation, field, etc.

Reply:
I've left a comment to inform that it's currently just custom Cypher fields, but I think the rewording is better!